src/builtins/core/datetime.rs

```diff
@@ -285,14 +285,16 @@ impl PlainDateTime {
         other: &Self,
         options: ResolvedRoundingOptions,
     ) -> TemporalResult<NormalizedDurationRecord> {
-        // 1. Assert: IsValidISODate(y1, mon1, d1) is true.
-        // 2. Assert: IsValidISODate(y2, mon2, d2) is true.
-        // 3. If CompareISODateTime(y1, mon1, d1, h1, min1, s1, ms1, mus1, ns1, y2, mon2, d2, h2, min2, s2, ms2, mus2, ns2) = 0, then
+        // 1. If CompareISODateTime(y1, mon1, d1, h1, min1, s1, ms1, mus1, ns1, y2, mon2, d2, h2, min2, s2, ms2, mus2, ns2) = 0, then
         if matches!(self.iso.cmp(&other.iso), Ordering::Equal) {
             // a. Let durationRecord be CreateDurationRecord(0, 0, 0, 0, 0, 0, 0, 0, 0, 0).
             // b. Return the Record { [[DurationRecord]]: durationRecord, [[Total]]: 0 }.
             return Ok(NormalizedDurationRecord::default());
         }
+        // 2. If ISODateTimeWithinLimits(isoDateTime1) is false or ISODateTimeWithinLimits(isoDateTime2) is false, throw a RangeError exception.
+        self.iso.check_validity()?;
```

Comment:
Double checked where this is called and this definitely needs to be added. Although, I'm a bit curious why the validity checks on that duration operation are in `diff_datetime_with_rounding` vs. inlined.

Reply:
It's because if the datetimes are out-of-bounds-but-equal it's still valid. When I first tried to fix this I started failing other tests.

Reply:
Huh, good to know.

I wonder if it would be worthwhile having a new type for ISO date records that haven't been validated yet.